crates/runtime/src/lib.rs

```diff
@@ -291,6 +385,26 @@ pub trait ExtensionLogic {
     ) -> Result<CheatcodeHandlingResult, EnhancedHintError> {
         Ok(CheatcodeHandlingResult::Forwarded)
     }
+
+    /// Different from `override_system_call` because it cannot be overridden,
+    /// always receives a signal and cannot return an error
+    fn listen_system_call(
```

Comment:
those names are somewhat wrong since it does not listen, but triggers a handler

Reply:
Done